### ISSUE TITLE
mongodb: 3.3.2 driver unable to connect with replica set when one node is down.

### ISSUE THREAD

**Issue by yog27ray** [parse-community/parse-server#6261]
MongoDB driver 3.3.2 is unable to connect to mongodb replicaSet if one of member machine is down.

Note: Same issue is described in https://jira.mongodb.org/browse/NODE-2231

### Steps to reproduce
1. Create mongodb replicaSet with 3 member.
2. connect parse-server with mongodb providing all 3 member host name in uri.
3. stop PRIMARY mongodb server
4. MongoDB is able to select new PRIMARY from the available 2 member but parser-server is unable to connect to new PRIMARY mongodb node. 

Note: Even we restart parse-server when one of the 3 member replicaSet mongodb is down and in mongo connection URI we have provided hostname for all the 3 member. It is not able to connect mongodb.
 
### Expected Results
When one mongoDB replicaSet member goes down. It should automatically connect to new primary mongodb server.

### Actual Outcome

Error: connect ECONNREFUSED x.x.x.x:port
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1128:14) {
  name: 'MongoNetworkError',
  errorLabels: [ 'TransientTransactionError' ],
  [Symbol(mongoErrorContextSymbol)]: {}
}

### Environment Setup

- **Server**
  - parse-server : 3.9.0
  - Operating System:  Ubuntu 18.04
  - Hardware:  R4.2x.large
  - Localhost or remote server? (AWS, Heroku, Azure, Digital Ocean, etc): AWS

- **Database**
  - MongoDB version: 4.2 (also tested on 4.0.9)
  - Storage engine:  WireTigger
  - Hardware:  R4.2x.large
  - Localhost or remote server? (AWS, mLab, ObjectRocket, Digital Ocean, etc): AWS

**Comment by manjeshpv** [comment]
I'm also facing same issue

**Comment by davimacedo** [comment]
As peer as the link you sent, it looks that updating the MongoDB package to the latest version should fix the problem. @dplewis why did you close https://github.com/parse-community/parse-server/pull/6242? Didn't it pass the tests?

**Comment by dplewis** [comment]
https://github.com/parse-community/parse-server/pull/6135#issuecomment-547065385

The test have been failing, it should be an easy fix. I can look into it again shortly

**Comment by davimacedo** [comment]
Sorry. I missed that one. Let me know if you need any help.

**Comment by dplewis** [comment]
@yog27ray  Can you try to see if the latest mongo driver update fixed your issue?

I should have a replica set somewhere to test with.

**Comment by yog27ray** [comment]
@dplewis latest mongo driver has fixed my issue. 

Below are the steps I used to confirm that latest mongo driver is fixing the issue.

1. deleted  mongodb driver 3.3.2 folder located at "node_modules/parse-server/node_modules/mongodb"
2. copied mongodb driver 3.3.5 folder at  "node_modules/parse-server/node_modules/mongodb"
3. restarted parse-server

now even in stop any replica member, parse-server is able to switch connection to other replica member and even i restart parse-server while keeping one member down it is able to communicate with other replica member.

We already configured replica set to verify the same if you want we can share the same with you.
